### PR TITLE
Add searchbar and filtering for home page

### PR DESCRIPTION
**Pull request by ac-jorellanaf**
Title: `Add searchbar and filtering for home page`

Description:
```
- Add searchbar and filtering for home page
- closes GDEV-830
```

### CODE REVIEW

File: src/components/home/home.tsx [hunk 1]
```diff
@@ -53,20 +112,62 @@ const Home = () => {
...
+                            {y.option}
+                          </option>
+                        ))
+                    )}
+                </Form.Select>
+                <Form.Select
+                  className="d-inline-block w-25 fs-8 text-capitalize"
+                  size="sm"
+                  onChange={(event) =>
+                    filterChange("has_study.type", event.target.value)
```
Comment (alamgirmani) [inline src/components/home/home.tsx:140]: Change this to type

File: src/components/home/home.tsx [hunk 2]
```diff
@@ -53,20 +112,62 @@ const Home = () => {
...
-                <option>Study Type</option>
-              </Form.Select>
-              <Form.Select className="d-inline-block w-25">
-                <option>Type</option>
-              </Form.Select>
-              <Form.Select className="d-inline-block w-25">
-                <option>Size</option>
-              </Form.Select>
+              <div className="w-75">
+                <Form.Select
```
Comment (alamgirmani) [inline src/components/home/home.tsx:116]: Both the Form.Select code can be moved to a method and can be called for each of the facet type

File: src/components/home/home.tsx [hunk 3]
```diff
@@ -10,8 +10,62 @@ import {
...
+    }
+  }
+
+  const searchDatasets = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (filterDict.length > 0) {
+      let filterURI = filterDict.map((x) => x.key + ":" + x.value).join(";");
+      navigate("/browse?q=" + searchKeyword + "&f=" + filterURI);
+    } else {
+      navigate("/browse?q=" + searchKeyword);
```
Comment (alamgirmani) [inline src/components/home/home.tsx:55]: add p=1

File: src/components/home/home.tsx [hunk 4]
```diff
@@ -10,8 +10,62 @@ import {
...
+      facetList = [];
+      dsCount = 0;
+    }
+  }
+
+  const searchDatasets = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (filterDict.length > 0) {
+      let filterURI = filterDict.map((x) => x.key + ":" + x.value).join(";");
+      navigate("/browse?q=" + searchKeyword + "&f=" + filterURI);
```
Comment (alamgirmani) [inline src/components/home/home.tsx:53]: Check for empty search keyword and if true, do not add "q" to the search params in URL

File: src/components/home/home.tsx [hunk 5]
```diff
@@ -10,8 +10,80 @@ import {
...
+
+  const [filterDict, setFilterDict] = React.useState<facetFilterModel[]>([]);
+
+  React.useEffect(
+    () => {
+      const getData = () => {
+        getDatasetsSearchResp(setSearchResults, [], "", 0, 1);
+      };
+      getData();
+    }, // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (ac-jorellanaf) [reply src/components/home/home.tsx:32]: So we can get the list of facets to filter, as well as the number of datasets.

File: src/components/home/home.tsx (hunk 2 above)
Comment (alamgirmani) [reply src/components/home/home.tsx:134]: Once the Form.Select block is converted to a method, we can call it as many times needed even if there are more than 2 different facet types. The code is the same for both the facet types, just pass the key of the facet to the method as argument.

**Review by alamgirmani — APPROVED** [review]

**Review by alamgirmani — APPROVED** [review]

**Pull request merged**